contracts/orderbook/Keepers/OrderKeeperClear.sol

```diff
@@ -0,0 +1,45 @@
+pragma solidity ^0.8.0;
+import "../IOrderBook.sol";
+import "@openzeppelin-4.3.2/token/ERC20/extensions/IERC20Metadata.sol";
+import "../../governance/PausableGuardian_0_8.sol";
+
+contract OrderKeeperClear is PausableGuardian_0_8 {
+    IOrderBook public factory;
+
+    constructor(IOrderBook factoryAddress) {
+        factory = factoryAddress;
+    }
+
+    function checkUpKeep(bytes calldata checkData)
+        external
+        view
+        returns (bool upkeepNeeded, bytes memory performData)
+    {
+        (uint256 start, uint256 end) = abi.decode(checkData, (uint256, uint256));
+        uint256 orderIDLength = factory.getTotalActiveOrders();
+        if(end < orderIDLength) {
+            if (start > orderIDLength) {
+                end = orderIDLength;
+            } else {
+                return (upkeepNeeded, performData);
+            }
+        }
+        IOrderBook.Order[] memory listOfMainOrders = factory
+            .getOrdersLimited(start, end);
```

Comment:
I wonder if you make getClearOderList on factory. will that make things more efficient? this way you won't have to do for loop here and will just pass result to performData. if you make proper in/out you won't even have to abi.encode

Reply:
it might be. Will look into it